common_knowledge/ARCHITECTURE.md

```diff
@@ -0,0 +1,505 @@
+# Commonwealth Monorepo Architecture
+
+## 1. Overview
+
+The Commonwealth monorepo is a modular, event-driven platform for decentralized governance, social, and blockchain-based applications. It leverages a CQRS (Command Query Responsibility Segregation) pattern, a rich domain model, and a robust infrastructure stack to support scalable, extensible, and secure community features.
+
+---
+
+## 2. System Components
+
+### 2.1 Packages
+
+- **commonwealth**: Main application (API server, client, scripts, deployment logic)
+- **farcaster-app**: Farcaster protocol integration
+- **snapshot-listener**: Listens to and processes Snapshot events
+- **load-testing**: Load and performance testing utilities
+- **external-api-testing**: Utilities for testing external APIs
+
+### 2.2 Libraries
+
+- **model**: Domain logic, aggregates, policies, services, and DB models
+- **core**: Framework utilities, CQRS, types, error handling
+- **adapters**: Protocol adapters (trpc, express, rabbitmq, etc)
+- **schemas**: Zod schemas for API and DB validation
+- **shared**: Shared utilities, constants, and types
+- **evm-protocols**: EVM-compatible blockchain integrations
+- **chains**: Chain adapters for various protocols
+- **railway, sitemaps, eslint-plugin**: Miscellaneous utilities and plugins
+
+### 2.3 Workers/Jobs
+
+- **graphileWorker**: Background job processing
+- **knock**: Notification and messaging jobs
+- **twitterWorker**: Twitter event processing
+- **evmChainEvents**: EVM chain event listeners
+- **commonwealthConsumer**: Consumes and processes RabbitMQ messages
+- **messageRelayer**: Relays messages between services
+- **discordBot**: Discord bot integration
+
+---
+
+## 3. Application Flow
+
+### 3.1 Client-Server Interaction
+
+- React web client communicates with the API server via HTTP and WebSocket.
+
+### 3.2 API Layer
+
+- Node.js/Express server exposes REST, TRPC, and WebSocket endpoints.
+
+### 3.3 Domain Logic
+
+- Implemented in the `model` lib using CQRS, aggregates, policies, and services.
+
+### 3.4 Adapters
+
+- Protocol adapters connect core logic to external systems (RabbitMQ, Express, TRPC, etc).
+
+---
+
+## 4. Infrastructure
+
+### 4.1 Databases
+
+- PostgreSQL for persistent storage. See [Database-ERD.md](Database-ERD.md) for schema.
```

Comment:
Database-ERD.md is probably super out of date.

Reply:
we should run the update script